openmc/stats/multivariate.py

```diff
@@ -0,0 +1,409 @@
+from abc import ABCMeta, abstractmethod
+from collections import Iterable
+from math import pi
+from numbers import Real
+import sys
+from xml.etree import ElementTree as ET
+
+import numpy as np
+from numpy.linalg import norm
+
+import openmc.checkvalue as cv
+from openmc.stats.univariate import Univariate, Uniform
+
+if sys.version_info[0] >= 3:
+    basestring = str
+
+
+class UnitSphere(object):
+    """Distribution of points on the unit sphere.
+
+    This abstract class is used for angular distributions, since a direction is
+    represented as a unit vector (i.e., vector on the unit sphere).
+
+    Parameters
+    ----------
+    name : str
+        Name of the distribution
+    reference_uvw : Iterable of Real
+        Direction from which polar angle is measured
+
+    Attributes
+    ----------
+    name : str
+        Name of the distribution
+    reference_uvw : Iterable of Real
+        Direction from which polar angle is measured
+
+    """
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self, name, reference_uvw=None):
+        self.name = name
+        self._reference_uvw = None
+        if reference_uvw is not None:
+            self.reference_uvw = reference_uvw
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def reference_uvw(self):
+        return self._reference_uvw
+
+    @name.setter
+    def name(self, name):
+        cv.check_type('name', name, basestring)
+        self._name = name
+
+    @reference_uvw.setter
+    def reference_uvw(self, uvw):
+        cv.check_type('reference direction', uvw, Iterable, Real)
+        uvw = np.asarray(uvw)
+        self._reference_uvw = uvw/norm(uvw)
+
+    @abstractmethod
+    def to_xml(self):
+        return ''
+
+
+class PolarAzimuthal(UnitSphere):
+    """Angular distribution represented by polar and azimuthal angles
+
+    This distribution allows one to specify the distribution of the cosine of
+    the polar angle and the azimuthal angle independently of once another.
+
+    Parameters
+    ----------
+    mu : openmc.stats.Univariate
+        Distribution of the cosine of the polar angle
+    phi : openmc.stats.Univariate
+        Distribution of the azimuthal angle
+    name : str, optional
+        Name of the distribution. Defaults to 'angle'.
+    reference_uvw : Iterable of Real
+        Direction from which polar angle is measured. Defaults to the positive
+        z-direction.
+
+    Attributes
+    ----------
+    mu : openmc.stats.Univariate
+        Distribution of the cosine of the polar angle
+    phi : openmc.stats.Univariate
+        Distribution of the azimuthal angle
+
+    """
+
+    def __init__(self, mu=None, phi=None, name='angle',reference_uvw=[0., 0., 1.]):
```

Comment:
Space between , and "reference" (or better yet, a newline to put it under 80 columns... ya know, just in case we ever decide to punch out our Python API on Hollerith cards.)